core/src/com/unciv/ui/screens/pickerscreens/ImprovementPickerScreen.kt

```diff
@@ -163,6 +163,27 @@ class ImprovementPickerScreen(
                 tile.getCity(),
                 cityUniqueCache
             )
+
+            //Determine if a stat will be increased by the current improvement
+            var isStatIncrease = false;
+            for (stat in stats) {
+                if (stat.value > 0.0) {
+                    isStatIncrease = true
+                    break
+                }
+            }
+
+            //Warn when the current improvement will increase a stat for the tile,
+            // but the tile is outside of the range (> 3 tiles from city center) that can be
+            // worked by a city's population
+            if (isStatIncrease
+                    && !improvement.name.startsWith(Constants.remove)
+                    && !improvement.isRoad()
+                    && tile.owningCity != null
+                    && !tile.owningCity!!.getWorkableTiles().contains(tile)
```

Comment:
My understanding of your comment is that, for instance, a city has only expanded cultural borders 2 tiles from the city center. You are concerned that the warning would appear for the third tile away, when that tile would be workable in the future.

If my above understanding is correct, the warning only shows when the tile is inside the cultural borders. Otherwise, the player gets the superseding "Build inside your cultural borders" message.